apps/nextjs-website/src/lib/strapi/codecs/BlocksContentCodec.ts

```diff
@@ -0,0 +1,115 @@
+import * as t from 'io-ts/lib';
+import { type BlocksContent } from '@strapi/blocks-react-renderer';
+
+const TextInlineNodeCodec = t.intersection([
+  t.strict({
+    type: t.literal('text'),
+    text: t.string,
+  }),
+  t.partial({
+    bold: t.boolean,
+    italic: t.boolean,
+    underline: t.boolean,
+    strikethrough: t.boolean,
+    code: t.boolean,
+  }),
```

Comment:
I can't find which use case the `partial` covers. If I'm not wrong, Strapi always returns properties, even if a property is not populated. In such cases, Strapi returns the property evaluated to null

Reply:
That's not the case. If the type returned by Strapi is `blocks`, then these fields could be populated or not.
This is an example returned by Strapi:
```json
"subhead": [
            {
              "type": "paragraph",
              "children": [
                { "type": "text", "text": "Ciao " },
                { "text": "ciao ", "type": "text", "bold": true },
                { "text": "ciao ", "type": "text", "italic": true },
                { "text": "ciao", "type": "text", "underline": true },
                { "text": " ", "type": "text" },
                { "text": "ciao", "type": "text", "strikethrough": true },
                { "text": " ", "type": "text" },
                { "text": "ciao", "type": "text", "code": true },
                { "text": " ciao", "type": "text" }
              ]
            }
          ],
```